### PR TITLE
Repo unit test

### PR DESCRIPTION
**Pull request by manuranga**
1) Making repo location configurable
2) Adding basic unit test for User Repo

### CODE REVIEW

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/packaging/DependentPackageTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package org.ballerinalang.test.packaging;
```
Comment (anupama-pathirage) [inline modules/ballerina-test/src/test/java/org/ballerinalang/test/packaging/DependentPackageTest.java]: Add License header

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/packaging/DependentPackageTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import org.ballerinalang.packerina.UserRepositoryUtils;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class DependentPackageTest extends PackagingTest {
+
+
```
Comment (anupama-pathirage) [inline modules/ballerina-test/src/test/java/org/ballerinalang/test/packaging/DependentPackageTest.java]: Unnecessary new line?

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/packaging/PackagingTest.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package org.ballerinalang.test.packaging;
```
Comment (anupama-pathirage) [inline modules/ballerina-test/src/test/java/org/ballerinalang/test/packaging/PackagingTest.java]: Add License header

File: cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/util/BRunUtil.java [hunk 4]
```diff
@@ -176,7 +177,8 @@ protected static void invokePackageInit(CompileResult compileResult, String pack
      */
     public static BValue[] invoke(CompileResult compileResult, String functionName, BValue[] args) {
         if (compileResult.getErrorCount() > 0) {
-            throw new IllegalStateException("compilation contains errors.");
+            throw new IllegalStateException("compilation contains errors." +
+                    Arrays.toString(compileResult.getDiagnostics()));
```
Comment (sameera-jayasoma) [inline cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/util/BRunUtil.java]: I think we print error before this method. Please check.

**Comment by hasithaa** [comment]
@manuranga Do we still need this PR? If so, please re-open PR and resolve conflits.

**Pull request closed without merging**